src/app/etl/generate_perf_report.py

```diff
@@ -0,0 +1,126 @@
+import polars as pl
+from typing import Optional
+from src.base.base_model import BaseModel
+from src.helpers.db_utils import execute_query
+from src.helpers.logger import LoggerFactory
+
+from src.helpers.utils import read_file_as_string
+
+logger = LoggerFactory.get_logger(__name__)
+
+class GeneratePerfReport(BaseModel):
+    """
+    Class for generating performance reports for funds.
+
+    This class analyzes fund data to identify the best performing funds for each month
+    based on rate of return calculations. It generates an Excel report with the results.
+    """
+
+    def __init__(self, app_config):
+        """
+        Initialize the GeneratePerfReport class.
+
+        Args:
+            app_config (AppConfig): Application configuration object containing paths and settings
+        """
+        super().__init__(app_config)
+        self.config = app_config
+        self.db_path = self.config.db_path
+        self.funds_folder = self.config.funds_folder
+        self.sql_path_pub_reference_data = self.config.sql_query_get_pub_reference_data
+        self.sql_path_pub_funds_equities_data = self.config.sql_query_get_pub_funds_equities_data
+        self.report_perf_path = self.config.report_perf_path
+        self.sql_path_active_funds_cfg = self.config.sql_query_get_active_funds_cfg
+        self.col_fund_name = "FUND NAME"
+
+    def _compute(self) -> None:
+        """
+        Execute the performance report generation process.
+
+        This method is called by the BaseModel's run method and serves as the main
+        entry point for the report generation process.
+
+        Returns:
+            None
+        """
+        self.generate_best_performing_fund_report()
+
+    def get_funds_data(self) -> pl.DataFrame:
```

Comment:
If the parameters from `self.config` and `self.sql_path_pub_funds_equities_data` were passed in explicitly as arguments to the method, then a staticmethod (or classmethod) could be used, making testing easier.